contracts/base/ExecutionHelper.sol

```diff
@@ -64,11 +72,20 @@ contract ExecutionHelper is IExecutionHelperEventsAndErrors {
         assembly {
             result := mload(0x40)
             calldatacopy(result, callData.offset, callData.length)
-            success := call(gas(), target, value, result, callData.length, codesize(), 0x00)
-            mstore(result, returndatasize()) // Store the length.
-            let o := add(result, 0x20)
-            returndatacopy(o, 0x00, returndatasize()) // Copy the returndata.
-            mstore(0x40, add(o, returndatasize())) // Allocate the memory.
+
+            // Check if the target has code
+            if iszero(extcodesize(target)) {
+                success := 0
+                mstore(result, 0) // Set result length to 0
+                mstore(0x40, add(result, 0x20)) // Update free memory pointer
+            } 
+            if iszero(success) {
```

Comment:
success would already be 0 if it doesn't go to previous block.

Reply:
then it will still do this for EOA

 success := call(gas(), target, value, result, callData.length, codesize(), 0x00)